crawler.py

```diff
@@ -102,14 +102,26 @@ def scrape_pages(self, aspect_list):
                 response.close()
                 if response.status is 200:
                     save_html(body, url)
-                    print("Finished crawling " + url)
+                    print("Finished crawling ", url)
+                else:
+                    print(response.status)
 
 
 def save_html(html, page):
-    print(page)
+
+    warning = """
+            <div style="position:fixed;    bottom:0;left:0;    border-top-right-radius: 8px;    color:  white;
+            background-color: red;  padding: .5em;">
+                This is a read-only mirror of the OSF.
+            </div>
+            """
+    sliding_footer = """<div id="footerSlideIn" style="display: block;">"""
+    no_sliding_footer = """<div id="footerSlideIn" style="display: none;">"""
+
+    html = html.decode('utf-8').replace('</body>', warning + '</body>').replace(sliding_footer, no_sliding_footer)
```

Comment:
We can definitely forgo removing the pop-up (it only shows on analytics).
The mirror warning can be concatenated to the end instead of replacing and
still work, but I didn't want it to be too shifty.

On Tue, Jun 14, 2016 at 9:05 AM, Yuhuai Liu <notifications@github.com>
wrote:

> In crawler.py
> <https://github.com/zamattiac/ROSIEBot/pull/9#discussion_r66965185>:
>
> >
> >
> >  def save_html(html, page):
> > -    print(page)
> > +
> > +    warning = """
> > +            <div style="position:fixed;    bottom:0;left:0;    border-top-right-radius: 8px;    color:  white;
> > +            background-color: red;  padding: .5em;">
> > +                This is a read-only mirror of the OSF.
> > +            </div>
> > +            """
> > +    sliding_footer = """<div id="footerSlideIn" style="display: block;">"""
> > +    no_sliding_footer = """<div id="footerSlideIn" style="display: none;">"""
> > +
> > +    html = html.decode('utf-8').replace('</body>', warning + '</body>').replace(sliding_footer, no_sliding_footer)
>
> Would there be a performance issue since the replace() method would
> probably do a string comparison and the html decoded seems to be quite a
> long string.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zamattiac/ROSIEBot/pull/9/files/a177bd9bb01e1fcfe25ef05e7a3c20da95a90e65#r66965185>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHeVeIs3ea1PlQuu3N7As7aTkeg5SgXks5qLqcggaJpZM4I0so0>
> .
>



-- 
Matthew Keitelman
571-269-4724
mak2vr@virginia.edu